app/uk/gov/hmrc/gform/testonly/TestOnlyController.scala

```diff
@@ -528,18 +528,37 @@ class TestOnlyController(
   }
 
   def restoreAll(formTemplateId: FormTemplateId, maybeAccessCode: Option[AccessCode]) =
+    controllerComponents.actionBuilder.async { implicit request =>
+      val maybeSnapshotId = request.body.asFormUrlEncoded.get("snapshotId").headOption
```

Comment:
I can see you are using this `.head` on multiple places. Would it be possible to improve it everywhere? 

Ideally we should be using `play.api.data.Form` to bind these forms (instead of low level `request.body.asFormUrlEncoded`). That should give us way to handle errors properly also.

For example edit snapshot data page is easy to crash.